### DISCUSSION TITLE
SO 2.3.10 Osquery agents do not reconnect to fleet after reboot

### DISCUSSION THREAD

**Discussion by Jakes-Lab** [Security-Onion-Solutions/securityonion#2050] [Unsupported Versions]
I haven't tested windows osquery agents, but on systems running Ubuntu 20.04 after reboot the endpoint will not reconnect to Fleet and the package deb-launcher must be re-installed.

**Comment by defensivedepth** [comment]
Do you have any agents that are still in a failed state? If so, can you post the sanitized logs so that we can figure out the root cause?

`systemctl status launcher.so-launcher.service -l`

Thanks!

**Reply by Jakes-Lab** [reply]
``launcher.so-launcher.service - The Kolide Launcher
     Loaded: loaded (/lib/systemd/system/launcher.so-launcher.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2020-11-25 14:27:04 UTC; 6h left
   Main PID: 1072 (launcher)
      Tasks: 35 (limit: 11855)
     Memory: 65.8M
     CGroup: /system.slice/launcher.so-launcher.service
             ├─1072 /usr/local/so-launcher/bin/launcher -config /etc/so-launcher/launcher.flags
             ├─1231 /usr/local/so-launcher/bin/osqueryd-updates/1605666031/osqueryd --logger_plugin=kolide_grpc --distributed_plugin=kolide_grpc --disable_distributed=false --distributed_interval=5 --pack_delimiter=: --host_identifier=u>
             └─1248 /usr/local/so-launcher/bin/osquery-extension.ext --socket /var/so-launcher/securityonion/osquery.sock --timeout 10 --interval 3

Nov 25 14:27:05 games1 launcher[1072]: {"caller":"query_target_updater.go:21","msg":"query target updater started","severity":"info","ts":"2020-11-25T14:27:05.141862014Z"}
Nov 25 14:27:05 games1 launcher[1072]: {"caller":"updater.go:74","msg":"updater started","severity":"info","ts":"2020-11-25T14:27:05.142065245Z"}
Nov 25 14:27:05 games1 launcher[1072]: {"arg0":"/usr/local/so-launcher/bin/osqueryd-updates/1605666031/osqueryd","args":"/usr/local/so-launcher/bin/osqueryd-updates/1605666031/osqueryd --logger_plugin=kolide_grpc --distributed_plugin=ko>
Nov 25 14:27:05 games1 osqueryd[1231]: osqueryd started [version=4.5.1]
Nov 25 14:27:05 games1 launcher[1072]: {"caller":"","component":"osquery","level":"stderr","msg":"[/usr/local/so-launcher/bin/osquery-extension.ext --socket /var/so-launcher/securityonion/osquery.sock --timeout 10 --interval 3]","severi>
Nov 25 14:27:05 games1 launcher[1072]: {"caller":"extension.go:128","msg":"extension started","severity":"info","ts":"2020-11-25T14:27:05.693219135Z"}
Nov 25 14:27:05 games1 launcher[1072]: {"caller":"interface.cpp:110","component":"osquery","level":"stderr","msg":"I1125 14:27:05.695793  1281 interface.cpp:110] Registering extension (kolide, 37629, version=, sdk=)","severity":"info",">
Nov 25 14:27:05 games1 launcher[1072]: {"caller":"options.cpp:92","component":"osquery","level":"stderr","msg":"W1125 14:27:05.709581  1231 options.cpp:92] Cannot set unknown or invalid flag: enable_windows_events_publisher","severity":>
Nov 25 14:27:05 games1 launcher[1072]: {"caller":"options.cpp:92","component":"osquery","level":"stderr","msg":"W1125 14:27:05.709617  1231 options.cpp:92] Cannot set unknown or invalid flag: enable_windows_events_subscriber","severity">
Nov 25 14:27:05 games1 launcher[1072]: {"caller":"events.cpp:867","component":"osquery","level":"stderr","msg":"I1125 14:27:05.710937  1231 events.cpp:867] Event publisher not enabled: auditeventpublisher: Publisher disabled via configu>
lines 1-21/21 (END)
``
